File: 0704-binary-search/0704-binary-search.cpp
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int left = 0;
        int right = nums.size() - 1;

        while (left <= right)
        {
            int mid = (left + right) / 2;
            int midn = nums[mid];
            if (midn == target)
            {
                return mid;
            }
            else if (midn < target)
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let left = 0;
  let right = nums.length - 1;

  while (left <= right) {
    let mid = Math.floor((left + right) / 2);
    let midN = nums[mid];
    if (midN === target) {
      return mid;
    } else if (midN < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }
  return -1;
};